src/Scripting/Core/Hosting/CommandLine/CommandLineRunner.cs

```diff
@@ -174,10 +174,11 @@ private int RunScript(ScriptOptions options, string code, ErrorLogger errorLogge
             var script = Script.CreateInitialScript<object>(_scriptCompiler, code, options, globals.GetType(), assemblyLoaderOpt: null);
             try
             {
-                script.RunAsync(globals, cancellationToken).Wait();
+                var task = script.RunAsync(globals, cancellationToken);
 
-                // TODO: use the return value of the script https://github.com/dotnet/roslyn/issues/5773
-                return CommonCompiler.Succeeded;
+                // Use the return value of the script if it's an int.
+                var returnValue = task.Result.ReturnValue;
+                return (returnValue is int) ? (int)returnValue : CommonCompiler.Succeeded;
```

Comment:
I think it's fine to keep it as ```int``` (I think that's all you're allowed to specify on ```void Main```).